packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+    // Top line of chart.
+    canvas.drawLine(
+      Offset(
+        chartController.xCanvasChart,
+        chartController.yCanvasChart,
+      ),
+      Offset(
+        chartController.canvasChartWidth +
+            chartController.xCanvasChart -
+            chartController.xPaddingRight,
+        chartController.yCanvasChart,
+      ),
+      axis,
+    );
+
+    // Left-side of chart
+    canvas.drawLine(
+      Offset(
+        chartController.xCanvasChart,
+        chartController.yCanvasChart,
+      ),
+      Offset(
+        chartController.xCanvasChart,
+        chartController.yCanvasChart + chartController.canvasChartHeight,
+      ),
+      axis,
+    );
+
+    // Bottom line of chart.
+    canvas.drawLine(
+      Offset(
+        chartController.xCanvasChart,
+        chartController.yCanvasChart + chartController.canvasChartHeight,
+      ),
+      Offset(
+        chartController.canvasChartWidth +
+            chartController.xCanvasChart -
+            chartController.xPaddingRight,
+        chartController.yCanvasChart + chartController.canvasChartHeight,
+      ),
+      axis,
+    );
+  }
+
+  /// Separated out from drawAxis because we don't know range until plotted.
+  // TODO(terry): Consider combining into drawAxis function.
+  void drawYTicks(Canvas canvas, ChartController chartController, Paint axis) {
+    final xCoord = chartController.xCanvasChart;
+
+    final yScale = chartController.yScale;
+
+    for (var labelIndex = yScale.labelTicks; labelIndex >= 0; labelIndex--) {
+      final unit = pow(10, yScale?.labelUnitExponent).toInt();
+      final y = labelIndex * unit;
+      final yCoord = chartController.yPositonToYCanvasCoord(y);
+
+      final labelName = constructLabel(
+        labelIndex.toInt(),
+        yScale.labelUnitExponent.toInt(),
+      );
+
+      drawText('$labelName', canvas, 5, yCoord);
+
+      // Draw horizontal tick.
+      canvas.drawLine(
+        Offset(xCoord, yCoord),
+        Offset(xCoord - 6, yCoord),
+        axis,
+      );
+    }
+  }
+
+  static String constructLabel(int labelValue, int unitExponent) {
+    var unit = '';
+    switch (unitExponent) {
+      case 1:
+      case 2:
+      case 3:
+        labelValue = labelValue * pow(10, unitExponent);
+        break;
+      // 10K, 100K
+      case 4:
+      case 5:
+        labelValue = labelValue * pow(10, unitExponent - 4);
+        unit = 'K';
+        break;
+      // 1M, 10M, 100M
+      case 6:
+      case 7:
+      case 8:
+        labelValue = labelValue * pow(10, unitExponent - 6);
+        unit = 'M';
+        break;
+      // 1B , 10B, 100B
+      case 9:
+      case 10:
+      case 11:
+        labelValue = labelValue * pow(10, unitExponent - 9);
+        unit = 'B';
+        break;
+      // 1T, 10T, 100T
+      case 12:
+      case 13:
+      case 14:
+        labelValue = labelValue * pow(10, unitExponent - 12);
+        unit = 'T';
+        break;
+    }
```

Comment:
is there prior art here? we might want to show 3000T instead of showing 3000000000000000 but probably won't matter in practice

Reply:
Was missing default case, for now I'll return e notation if we need values over trillion then I'll add those too.

        unit = 'e+$unitExponent';